### ISSUE TITLE
operators are not correctly handled in Expr pattern

### ISSUE THREAD

**Issue by Roger-luo** [thautwarm/MLStyle.jl#74]
it should be valid to use operators as function, but the instance of operator is not treated correctly

e.g this errors

```jl
function foo(ex)
    @match ex begin
        Expr(:call, sin, Expr(:call, *, 2, π)) => "2x"
        _ => "unknown"
    end
end
```

**Comment by Roger-luo** [comment]
So it seems I should write

```jl
function foo(ex)
    @match ex begin
        Expr(:call, sin, &(2π)) => "2x"
        Expr(:call, &*, 2, _) => "2x"
        _ => "unknown"
    end
end
```

however this makes instances of operators not function (their behaviour are different with functions)

**Comment by thautwarm** [comment]
```julia
function foo(ex)
           @match ex begin
               Expr(:call, sin, Expr(:call, :*, 2, :π)) => "2x" 
               _ => "unknown"
           end
       end
```

I think you want this?

Due to the `*` in an AST is a symbol before it's evaluated, you should treat it as a symbol as well.

**Comment by inkydragon** [comment]
Some inconsistent?

```
julia> @match :(sin(1)) begin
           Expr(:call, :sin, 1) => "sin"
           _ => "unknown"
       end
"sin"

julia> @match :(sin(1)) begin
           Expr(:call, sin, 1) => "sin" # should throw an error?
           _ => "unknown"
       end
"sin"

julia>

julia> @match :(1*1) begin
           Expr(:call, :*, _, _) => "*"
           _ => "unknown"
       end
"*"

julia> @match :(1*1) begin
           Expr(:call, *, _, _) => "*"
           _ => "unknown"
       end
ERROR: LoadError: PatternUnsolvedException("invalid usage or unknown case *")
Stacktrace:
...
```

**Comment by thautwarm** [comment]
@inkydragon 

```
@match :(1*1) begin
           Expr(:call, *, _, _) => "*"
           _ => "unknown"
       end
```
shouldn't match for the function `*` shouldn't match a symbol `:*`.
